scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/EquationSolvingAlgorithms.java

```diff
@@ -0,0 +1,417 @@
+   * value y is in [y-yLow, y+yHigh].
+   * x0 is a first guess for x, it must be close to the solution to avoid diverging.
+   * It is considered that algorithm is diverging when the iterated value of x goes out of [dxLow, dxHigh].
+   */
+  public interface NewtonAlgorithm<T>{
+    RootFindingResult<T> findRoot(FunctionWithDerivative<T> f,
+                                  T x0,
+                                  T y,
+                                  T confidenceYLow,
+                                  T confidenceYHigh,
+                                  T dxLow,
+                                  T dxHigh,
+                                  int maxNbIterations) throws DivergenceException, ZeroDerivative,
+                                                              ExceededMaxIterationException;
+  }
+
+  public interface Function<T> {
+    T valueAt(T x);
+    boolean isApproximation();
+  }
+
+  public interface FunctionWithDerivative<T> extends Function<T> {
+    T derivativeAt(T x);
+  }
+
+  public static class ZeroDerivative extends Exception{
+    public ZeroDerivative() {}
+  }
+
+  public static class DivergenceException extends Exception{
+    History<?> history;
+
+    public DivergenceException(String errorMessage) {
+      super(errorMessage);
+    }
+    public DivergenceException(String errorMessage, History<?> history) {
+      super(errorMessage);
+      this.history = history;
+    }
+  }
+  public static class WrongBracketingException extends Exception{
+    public WrongBracketingException(String errorMessage) {
+      super(errorMessage);
+    }
+  }
+
+  public static class ExceededMaxIterationException extends Exception{
+    History<?> history;
+    public ExceededMaxIterationException() {
+      super();
+    }
+    public ExceededMaxIterationException(History<?> history) {
+      super();
+      this.history = history;
+    }
+  }
+
+  public static class NoSolutionException extends Exception{
+    public NoSolutionException() {
+      super();
+    }
+    public NoSolutionException(String errorMessage) {
+      super(errorMessage);
+    }
+  }
+
+  final static class BrentDoubleAlgorithm implements BracketingAlgorithm<Double> {
+    public RootFindingResult<Double> findRoot(
+        Function<Double> f,
+        Double bracket1,
+        Double bracket2,
+        Double y,
+        Double confidenceYLow,
+        Double confidenceYHigh,
+        Double dxLow,
+        Double dxHigh,
+        int maxNbIterations)
+    throws ExceededMaxIterationException, WrongBracketingException, NoSolutionException
+    {
+      if (bracket1 < dxLow || bracket2 > dxHigh) {
+        throw new WrongBracketingException("brackets are out of prescribed root domain");
+      }
+      if (Math.abs(confidenceYLow) - Math.abs(confidenceYHigh) < 1E-6) {
+        throw new NoSolutionException("To use Brent's method, you must provide equal lower and upper confidence bounds.");
```

Comment:
Judging from the text of the exception, I would expect the subtraction to be inside the `abs` call, and the comparison to be "greater than" instead of less than.

```suggestion
      if (Math.abs(confidenceYLow - confidenceYHigh) > 1E-6) {
        throw new NoSolutionException("To use Brent's method, you must provide equal lower and upper confidence bounds.");
```

Reply:
Absolutely. It has been changed.